spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreTemplate.java

```diff
@@ -77,23 +77,32 @@ public DatastoreTemplate(Datastore datastore,
 	}
 
 	@Override
-	public <T> void save(T instance) {
-		this.datastore.put(convertToEntity(instance));
+	public <T> T save(T instance) {
+		Entity saved = this.datastore.put(convertToEntity(instance));
+		return this.datastoreEntityConverter.read((Class<T>) instance.getClass(), saved);
 	}
 
 	@Override
 	public <T> void deleteById(Object id, Class<T> entityClass) {
 		this.datastore.delete(getKeyFromId(id, entityClass));
 	}
 
+	@Override
+	public <T> void deleteAllById(Iterable<?> ids, Class<T> entityClass) {
```

Comment:
We should return a count here too, since non-existing IDs will cause no-op.

Reply:
We won't be aware of how many are actually deleted, because the client lib's delete function returns void and literally says that deleting non-existing keys is "OK". So, it will not be possible to give a count unless we check separately to see how many of those IDs exist beforehand.

Reply:
done.